python/revng/daemon/__init__.py

```diff
@@ -58,7 +60,7 @@ async def status(request):
 
 def startup():
     global manager, startup_done
-    capi_initialize()
+    capi_initialize(signals_to_preserve=(signal.SIGINT, signal.SIGTERM))
```

Comment:
There's `make_manager` in `__init__.py` that's there exactly to make it nice for python REPL/scripting users
e.g.
```python
from revng.daemon import make_manager

manager = make_manager()
manager.produce(...)
```
The capi stuff is handled manually in the daemon since the initialize and shutdown are handled within the server's lifecycle hooks

Reply:
So we have two `make_manager`, one that does `capi_initialize` and one that doesn't?
It's quite confusing.

I see two options:
1. We do not support having multiple managers: we have a single `make_manager` that does `capi_{initialize,shutdown}`.
2. We do support having multiple managers: we need `revng.Context` that offers a `make_manager` method and handles `capi_{initialize,shutdown}`.

Reply:
> So we have two make_manager, one that does capi_initialize and one that doesn't?

No

There's two ways of doing things:
1. You go with the "easy wrapper" and we handle stuff automatically, this is `revng.api.make_manager`
2. You want to handle the lifecycle manually (e.g. decide when to `initialize`, for instance in the daemon), then you call the `capi_{initialize,shutdown}` manually and construct the manager manually with `Manager(...)`

Reply:
This situation is confusing. Let's discuss this in person.